docs/src/research/installation.md

```diff
@@ -0,0 +1,60 @@
+
+For completeness - NixOS usage is popular in cloud system installations. Many cloud providers do not provide NixOS as option and bare-metal cloud is always not an option. For this need, approaches like [nixos-anywhere](https://github.com/nix-community/nixos-anywhere) have been developed. [Using smart approach with `kexec`](https://numtide.com/blog/we-dont-need-nixos-cloud-images-anymore-2/), one can completely replace cloud provider default Linux options.
+
+### Modular Interactive
+
+[Ghaf introduced a modular structure for interactive installer](https://tiiuae.github.io/ghaf/ref_impl/installer.html). The initial Ghaf reference installer still uses raw disk image per target device. In practice, just writes the raw disk image to the target device internal persitent memory - NVMe for example. The key idea with the modular interactive Ghaf installer is to enable customization of the installer per target device needs and at the same time support further development of Ghaf graphical installer. The challenge of the interactive installer is to amount of configurable options, development and testing them. Given the Ghaf approach of target device specific installation described in the last section of [Ghaf Initial Approach](installation.md#ghaf-initial-approach), the requirement for Ghaf device specific installer is challenging. Ghaf installer would have to either:
+
+* embed the device specific installation raw disk image in the installer (current way) which results in huge installer image
+* dynamically build the device specific installation according to the user interactive selection
+* download a pre-built device specific raw disk image which could result in huge number of configurations
+* use some combination of generic and specific (kind of a compromise)
+
+None of which seem feasible in the long run. None of which are either Ghaf objective in the long run.
+
+But how to achieve a device specific secure system installation without getting lost with the generic Linux distro requirements?
+
+### Declarative, Non-Interactive Installation
+
+Now that we already version control reference device specific secure system declarations, the question is if we can transfer those into the device installations without asking user too many questions that make the installation unnecessarily difficult to implement? This alone is not a novel idea. Automatic Original Equipment Manufacturer (OEM) installers have been doing this for long. Those are often not declarative but rather scripted guidance to answer questions in generic installers.
```

Comment:
Now that we already version control reference device-specific secure system declarations, the question is if we can transfer those into the device installations without asking a user too many questions that make the installation unnecessarily difficult to implement.

This alone is not a novel idea. Automatic Original Equipment Manufacturer (OEM) installers have been doing this for long. Those are often not declarative but rather scripted guidance to answer questions in generic installers.

Reply:
Done